src/providers/twitch/TwitchMessageBuilder.cpp

```diff
@@ -1190,10 +1189,23 @@ void TwitchMessageBuilder::appendTwitchBadges()
 
     auto badgeInfos = TwitchMessageBuilder::parseBadgeInfoTag(this->tags);
     auto badges = this->parseBadgeTag(this->tags);
+    TwitchMessageBuilder::appendTwitchBadges(this, badges, badgeInfos,
+                                             this->twitchChannel);
+}
+
+void TwitchMessageBuilder::appendTwitchBadges(
```

Comment:
It feels weird to have a static function accept a class like this of it's own base type. I think this function should instead just be defined in TwitchMessageBuilder but outside of the class
also nit: rename `TwitchChannel*` parameter to `twitchChannel`